source/game_sa/Text/KeyArray.h

```diff
@@ -19,7 +19,7 @@ class CKeyArray {
     void Unload();
 
     void /* inline */ Update(char* offset);
-    char* Search(const char* key, bool* found);
+    char* Search(const char* key, bool& found);
```

Comment:
```suggestion
    /*!
     * @brief Find text by it's key
     * 
     * @param     key   The text's key
     * @param out found Whenever the text was found
     *
     * @return The text identified by the given key or null if not found
    */
    char* Search(const char* key, bool& found);
```
Also, for some reason I suspect `entry->string` might be null too? Otherwise the `found` out var value makes no sense.